src/django_components/component.py

```diff
@@ -1798,344 +1846,373 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
         extensions._init_component_class(cls)
         extensions.on_component_class_created(OnComponentClassCreatedContext(cls))
 
-    @contextmanager
-    def _with_metadata(self, item: MetadataItem) -> Generator[None, None, None]:
-        self._metadata_stack.append(item)
-        yield
-        self._metadata_stack.pop()
+    ########################################
+    # INSTANCE PROPERTIES
```

Comment:
In this section below are the newly-assigned instance properties. For most part, these were previously property getters that took the value from the stack, and now they are plain attributes.